articles/building-a-nodejs-application-using-docker/index.md

```diff
@@ -0,0 +1,244 @@
+# Building A Node.js Application Using Docker
+
+[Docker](https://www.docker.com/) allows us to run our applications as *containers*. A container is a standalone executable package which is lightweight and has everything needed to run an application be it libraries, tools, runtime, settings or code.
+
+In this tutorial, we will build a Node.js application, create an image and also build a container using the image. Enjoy!
+
+
+## Dockerize the Node.js Application
+
+To start this tutorial, we need to make sure Docker and [Node.js](https://nodejs.org/en/download/package-manager/) is installed on our system. If Docker is not, you can refer to the links below to download:
+- [Ubuntu](https://docs.docker.com/engine/installation/linux/ubuntu/)
+- [Windows](https://docs.docker.com/docker-for-windows/install/)
+- [MacOS](https://docs.docker.com/docker-for-mac/install/)
+
+In order to dockerize a Node.js application, we need to follow these steps:
+
+- Creating A Node.js application
+- Create A Dockerfile
+- Building your Docker Image
+- Expose
+
+## Step 1 - Creating a Node.js Application
+
+First, we will start by creating a directory for our project and then install some dependencies for our simple Hello World website.
+```bash
+ mkdir node-web-app
+ cd node-web-app
+```
+
+## Install npm And Express Framework
+
+Install npm and express, which is a Node.js framework. Then, initialize npm in our directory.
+```
+ npm init
+```
+npm creates a `package.json` that holds the dependencies of the app. Next, install the express framework dependency.
+```
+ npm install express --save
+```
+Result:
+```js
+{
+  "name": "web-app",
+  "version": "1.0.0",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "express": "^4.17.1"
+  },
+  "devDependencies": {},
+  "description": ""
+}
+
+```
+Create a `app.js` file with an HTTP server that will set up the Hello World site:
+```js
+//Load express module with `require` directive
+var express = require('express')
+var app = express()
+
+//Define request response in root URL (/)
+app.get('/', function (req, res) {
+  res.send('Hello World!')
+})
+
+//Launch listening server on port 8080
+app.listen(8080, function () {
+  console.log('app listening on port 8080!')
+})
+```
+
+### Run The Application
+
+The app is ready to launch:
+```
+$ node app.js
+```
+Go to `http://localhost:8080/` in your browser to view it.
+
+## Step 2 - Create A DockerFile
+
+Create a file in the directory called `Dockerfile`.
```

Comment:
I would change this to root directory so readers don't think Dockerfile should be a directory.